### PR TITLE
Add Env Var support for access-token flag

### LINKED ISSUE #34: [Feature] Look for an INLETS_ACCESS_TOKEN_FILE env var if --access-token-file is not set
It would be nice to be able to set a env variable for the location of the access-token file. At the moment we need to set it for every command, and its lots to type

## Expected Behaviour

if you ommit --access-token-file and there is an environment variable set (`INLETS_ACCESS_TOKEN_FILE`) then we could try to use the value in that file (if it exists)

## Current Behaviour

you have to set it every time

## Possible Solution

check if the flag is set, if so we use the flag vlaue, if its not we could see if the env variable is set, and if it is we could use the value.

### PR DESCRIPTION
**Pull request by Waterdrips**
## Description

If the env variable INLETS_ACCESS_TOKEN or
INLETS_SECRET_KEY are set then these values will
be used AFTER the values for the flags. As in, if you
don't specify a flag it will see if these exist, if you
set the flag then it won't look in this env variable

Fixes #34 
Signed-off-by: Alistair Hey <alistair@heyal.co.uk>

## How Has This Been Tested?

unit tests (100% coverage of new code)

#### Ran the command with:

Env Var set, create & delete work.

Env Var set to something that doesnt exist
Create & delete need the --access-token flag set, and this flag overides the env var (command fail without the flav using the env var file)

Env var not set:
Flags needed

## How are existing users impacted? What migration steps/scripts do we need?
This is an additional flag for users so they dont have to pass `--access-token` or `--secret-key` with every command

## Checklist:

I have:

- [x] updated the documentation and/or roadmap (if required)
- [x] read the [CONTRIBUTION](https://github.com/inlets/inlets/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [x] added unit tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -183,6 +183,23 @@ inletsctl download --pro
...
 ```
 
+## Configuration using environment variables
+
+You may want to set an environment variable that points at your `access-token-file` or `secret-key-file`
+
+Inlets will look for the following:
+
+```sh
+# For providors that use --access-token-file
```
Comment (alexellis) [inline README.md]: Typo

File: pkg/env/env_test.go [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+	if err != nil {
+		t.Errorf("got an error, %s", err)
+	}
+
+	if want != got {
+		t.Errorf("want: %s but got: %s", want, got)
+	}
+}
+
+func makeTempFileWithRandomContents() (string, string) {
```
Comment (alexellis) [inline pkg/env/env_test.go]: Why do we need this? I'd rather we just wrote a file with canned contents.

File: pkg/env/env_test.go [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+	flags := pflag.FlagSet{}
+
+	flags.String("value-flag", value, "")
+
+	os.Setenv(envVarName, "BLANK VALUE")
+	got, err := GetRequiredFileOrString(&flags, "file-flag", "value-flag", envVarName)
+	os.Unsetenv(envVarName)
+
+	if err != nil {
+		t.Errorf("got error when getting value: %s", err.Error())
```
Comment (alexellis) [inline pkg/env/env_test.go]: Does this need a return?

File: pkg/env/env_test.go [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+	}
+
+	want := contents
+	if want != got {
+
+		t.Errorf("want: %s, but got: %s", want, got)
+	}
+}
+
+//
```
Comment (alexellis) [inline pkg/env/env_test.go]: Empty comment, delete or give a comment

File: pkg/env/env_test.go [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+	got, err := GetRequiredFileOrString(&flags, "file-flag", "value-flag", envVarName)
+	os.Unsetenv(envVarName)
+
+	if err != nil {
+		t.Errorf("got error when reading file: %s", err.Error())
+	}
+
+	want := contents
+	if want != got {
+
```
Comment (alexellis) [inline pkg/env/env_test.go]: Extra space not required

File: pkg/env/env.go [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+import (
+	"fmt"
+	"github.com/spf13/pflag"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+)
+
+func GetRequiredFileOrString(flags *pflag.FlagSet, file, value, envVarName string) (string, error) {
```
Comment (alexellis) [inline pkg/env/env.go]: Why are there two methods?

**Comment by alexellis** [comment]
Typo in the description - Add Env Var "suppoort"

**Comment by Waterdrips** [comment]
Thanks for the review - I have made changes based on the comments.

**Comment by alexellis** [comment]
If this pull request was merged how would you consume it on your dev machine?

I wonder if we doing the wrong thing and should be adding `export INLETS_TOKEN=$(cat my-token.txt)` to our bash profile instead.

**Comment by Waterdrips** [comment]
i am using this in my .bashrc (with my build of inlets on path)
```sh
export INLETS_ACCESS_TOKEN_FILE="~/do-token"
```
Your suggestion only works for single token providers, AWS needs key and secret key doesn't it?

**Comment by alexellis** [comment]
What I’m suggesting is that perhaps an ENV-var is required for the value, but isn’t required for a filename. Given how easy it is to read a value into an ENV in bash_profile etc.

But, perhaps I missed multi-line files? Do we even have multi-line tokens?

**Comment by Waterdrips** [comment]
Iv just looked, we only have single line files. Ill change to using the access key as env var and then put something in readme about getting the value from a file

**Comment by alexellis** [comment]
Thank you Alistair. I am in two minds about this, but I think we should try the less verbose option and see how it goes. This is a good PR to have, thank you 👍

**Comment by Waterdrips** [comment]
> Please change to be `INLETS_ACCESS_TOKEN` with an example in the README on how to populate in bash_profile etc.

@alexellis This has been changed

**Comment by alexellis** [comment]
Can you edit the description, title, and commit message if required as per the change?

**Comment by alexellis** [comment]
Screenshot 2020-02-06 at 10 35 51

**Comment by Waterdrips** [comment]
> Screenshot 2020-02-06 at 10 35 51

I did the commit but forgot about the github side. 

Should all be changed now

**Comment by alexellis** [comment]
Thank you Alistair, I think I'll start using this now.

**Pull request merged**